src/main/java/uk/nhs/digital/nhsconnect/lab/results/translator/mapper/ProcedureRequestMapper.java

```diff
@@ -1,12 +1,122 @@
 package uk.nhs.digital.nhsconnect.lab.results.translator.mapper;
 
+import lombok.AllArgsConstructor;
+import lombok.RequiredArgsConstructor;
+import org.hl7.fhir.dstu3.model.Annotation;
+import org.hl7.fhir.dstu3.model.CodeableConcept;
+import org.hl7.fhir.dstu3.model.Organization;
+import org.hl7.fhir.dstu3.model.Patient;
+import org.hl7.fhir.dstu3.model.Practitioner;
 import org.hl7.fhir.dstu3.model.ProcedureRequest;
+import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.stereotype.Component;
+import uk.nhs.digital.nhsconnect.lab.results.model.edifact.FreeTextSegment;
 import uk.nhs.digital.nhsconnect.lab.results.model.edifact.Message;
+import uk.nhs.digital.nhsconnect.lab.results.model.edifact.ReportStatusCode;
+import uk.nhs.digital.nhsconnect.lab.results.model.edifact.segmentgroup.PatientClinicalInfo;
+import uk.nhs.digital.nhsconnect.lab.results.utils.ResourceFullUrlGenerator;
+import uk.nhs.digital.nhsconnect.lab.results.utils.UUIDGenerator;
+
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.stream.Collectors;
+
+import org.hl7.fhir.dstu3.model.ProcedureRequest.ProcedureRequestStatus;
+import org.hl7.fhir.dstu3.model.ProcedureRequest.ProcedureRequestIntent;
 
 @Component
+@RequiredArgsConstructor(onConstructor_ = @Autowired)
 public class ProcedureRequestMapper {
-    public ProcedureRequest mapToProcedureRequest(final Message message) {
-        return new ProcedureRequest();
+    private final UUIDGenerator uuidGenerator;
+    private final ResourceFullUrlGenerator fullUrlGenerator;
+    private static final Map<ReportStatusCode, ProcedureRequestStatus> STATUS_CODE_MAPPING = Map.of(
+        ReportStatusCode.UNSPECIFIED, ProcedureRequestStatus.UNKNOWN
+    );
+
+    public Optional<ProcedureRequest> mapToProcedureRequest(
+        final Message message,
+        Patient patient,
+        Practitioner requestingPractitioner,
+        Organization requestingOrganization,
+        Practitioner performingPractitioner,
+        Organization performingOrganization
+    ) {
+        return new InternalMapper(message, patient, requestingPractitioner,
+            requestingOrganization, performingPractitioner, performingOrganization).map();
+    }
+
+    @AllArgsConstructor
+    private class InternalMapper {
+        private final Message message;
+        private final Patient patient;
+        private final Practitioner requestingPractitioner;
+        private final Organization requestingOrganization;
+        private final Practitioner performingPractitioner;
+        private final Organization performingOrganization;
+
+        public Optional<ProcedureRequest> map() {
+            return message.getServiceReportDetails().getSubject().getClinicalInfo()
+                .map(this::mapPatientClinicalInfo);
+        }
+
+        private ProcedureRequest mapPatientClinicalInfo(PatientClinicalInfo patientClinicalInfo) {
```

Comment:
The `PatientClinicalInfo` and `ProcedureRequest` could also become fields of this class, to really capitalise on the benefits the strategy provides. They'd be non-`final` and the annotation would need to be changed to `@RequiredArgsConstructor`.

Reply:
Sure, thanks for suggestion! Fixed in 35223bd82a5e7f261715c06c9b1818cef52873ff